benchmark/Main.hs

```diff
@@ -167,3 +169,14 @@ bgroupMember = do
                     , bench "HashSet/Foldable" $ whnf (Foldable.elem  sample) listHashSet
                     , bench "HashSet/elem"     $ whnf (Container.elem sample) listHashSet
                     ]
+
+-- | Checks that 'foldl'' is implemented efficiently for 'Universum.List'
+bgroupFold :: Benchmark
+bgroupFold = do
+    let testList        = [1..100000] :: [Int]
+    let universumFoldl' = Container.foldl' HashSet.insert mempty
+    let ghcFoldl'       = List.foldl' (\hashSet element -> 
+                          HashSet.insert element hashSet) mempty
+    bgroup "fold"   [ bench "List/foldl'" $ nf universumFoldl' testList
```

Comment:
Let's rename `bgroup "fold"` to `bgroup "foldl'"` and bench names from to `universum` and `base`. `foldl'` is already mentioned in `bgroup` so there's no need to write `foldl'` one more time. And `List` and `Base` are not clear enough names, as I think about it more...